platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/AddressBookInitializer.java

```diff
@@ -0,0 +1,274 @@
+    /** The SwirldState to use at genesis. */
+    @NonNull
+    private final Supplier<SwirldState> genesisSupplier;
+    /** The SignedState loaded from disk. May be null. */
+    @Nullable
+    private final SignedState loadedSignedState;
+    /** The address book in the signed state loaded from disk. May be null. */
+    @Nullable
+    private final AddressBook loadedAddressBook;
+    /** The address book derived from config.txt */
+    @NonNull
+    private final AddressBook configAddressBook;
+    /** The address book determined for use after {@link AddressBookInitializer#initialize()} is called. */
+    @NonNull
+    private final AddressBook initialAddressBook;
+    /** The path to the directory for writing address books. */
+    @NonNull
+    private final Path pathToAddressBookDirectory;
+    /** Indicate that the unmodified config address book must be used. */
+    private final boolean useConfigAddressBook;
+
+    /**
+     * Constructs an AddressBookInitializer to initialize an address book from the swirld application state, config.txt,
+     * and the saved state from disk.
+     *
+     * @param currentVersion       The current version of the application. Must not be null.
+     * @param signedState          The signed state loaded from disk.  May be null.
+     * @param genesisSupplier      The swirld application state in genesis start. Must not be null.
+     * @param configAddressBook    The address book derived from config.txt. Must not be null.
+     * @param parentDirectory      The parent directory of the address book directory. Must not be null.
+     * @param useConfigAddressBook Indicates if the unmodified config address book should be used.
+     */
+    public AddressBookInitializer(
+            @NonNull final SoftwareVersion currentVersion,
+            @Nullable final SignedState signedState,
+            @NonNull final Supplier<SwirldState> genesisSupplier,
+            @NonNull final AddressBook configAddressBook,
+            @NonNull final String parentDirectory,
+            @NonNull final boolean useConfigAddressBook) {
+        this.currentVersion = Objects.requireNonNull(currentVersion, "The currentVersion must not be null.");
+        this.genesisSupplier =
+                Objects.requireNonNull(genesisSupplier, "The genesis swirldState supplier must not be null.");
+        this.configAddressBook = Objects.requireNonNull(configAddressBook, "The configAddressBook must not be null.");
+        Objects.requireNonNull(parentDirectory, "The parentDirectory must not be null.");
+        this.useConfigAddressBook = useConfigAddressBook;
+        Path addressBookDirectoryPath = Path.of(parentDirectory, ADDRESS_BOOK_DIRECTORY_NAME);
+        try {
+            Files.createDirectories(addressBookDirectoryPath);
+        } catch (final IOException e) {
+            logger.error(EXCEPTION.getMarker(), "Not able to create directory: {}", addressBookDirectoryPath, e);
+            addressBookDirectoryPath = null;
```

Comment:
should we proceed with this being null?

Reply:
It is null only when there is an IOException.  Fail hard and throw a runtime exception?  IllegalState?

Reply:
Now throwing an IllegalStateException